src/Utilities/ContainerHelpers.hpp

```diff
@@ -0,0 +1,54 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "Utilities/ForceInline.hpp"
+#include "Utilities/Requires.hpp"
+
+/*!
+ * \ingroup UtilitiesGroup
+ * \brief Returns the `i`th element if `T` has a subscript operator, otherwise
+ * if `T` is fundamental returns `t`.
+ */
+template <class T>
+SPECTRE_ALWAYS_INLINE auto get_element(const T& t, const size_t i)
+    -> decltype(t[i]) {
+  return t[i];
+}
+
+/// \cond
+template <class T>
+SPECTRE_ALWAYS_INLINE auto get_element(T& t, const size_t i) -> decltype(t[i]) {
+  return t[i];
+}
+
+template <class T, Requires<std::is_fundamental<T>::value> = nullptr>
+SPECTRE_ALWAYS_INLINE const T& get_element(const T& t, const size_t /*i*/) {
+  return t;
+}
+
+template <class T, Requires<std::is_fundamental<T>::value> = nullptr>
+SPECTRE_ALWAYS_INLINE T& get_element(T& t, const size_t /*i*/) {
+  return t;
+}
+/// \endcond
+
+/*!
+ * \ingroup UtilitiesGroup
+ * \brief Retrieve the `size of `t` if `t.size()` is a valid expression,
+ * otherwise if `T` is fundamental returns 1
+ */
+template <class T>
+SPECTRE_ALWAYS_INLINE auto get_size(const T& t) -> decltype(t.size()) {
```

Comment:
I think this should return `size_t` so that `get_size(t)` is the same type for both cases.

Reply:
I think all stl containers return a `size_t` for `.size()` and all of our own containers have been trying to mimic that.

Reply:
Actually, the reason @gsb76 gave for why it should be `size_t` is the exact reason it shouldn't be. Calling this function should not produce a different return type that just calling `.size()` directly.